projects/packages/stats/src/class-xmlrpc.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * The Stats XMLRPC class
+ *
+ * @package automattic/jetpack-stats
+ */
+
+namespace Automattic\Jetpack\Stats;
+
+use Automattic\Jetpack\Constants;
+
+/**
+ * Stats XMLRPC.
+ *
+ * Adds additional methods to the WordPress XML-RPC API for handling Stats specific features.
+ *
+ * @since $$next-version$$
+ */
+class XMLRPC {
+
+	/**
+	 * Singleton XMLRPC instance.
+	 *
+	 * @var XMLRPC
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Private XMLRPC constructor.
+	 *
+	 * Use the XMLRPC::init() method to get an instance.
+	 */
+	private function __construct() {
+		add_filter( 'jetpack_xmlrpc_unauthenticated_methods', array( $this, 'xmlrpc_methods' ), 10, 3 );
+	}
+
+	/**
+	 * Initialize class and get back a singleton instance.
+	 *
+	 * @return XMLRPC
+	 */
+	public static function init() {
+		if ( null === self::$instance ) {
+			self::$instance = new XMLRPC();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds additional methods to the WordPress xmlrpc API for handling Stats specific features
+	 *
+	 * @param array $methods The Jetpack API methods.
+	 *
+	 * @return array
+	 */
+	public function xmlrpc_methods( $methods ) {
+
+		$methods['jetpack.getBlog'] = array( $this, 'get_blog' );
+
+		return $methods;
+	}
+
+	/**
+	 * Stats Get Blog.
+	 *
+	 * @return array
+	 */
+	public function get_blog() {
+		$home = wp_parse_url( trailingslashit( get_option( 'home' ) ) );
+		$blog = array(
+			'host'                => $home['host'],
+			'path'                => $home['path'],
+			'blogname'            => get_option( 'blogname' ),
+			'blogdescription'     => get_option( 'blogdescription' ),
+			'siteurl'             => get_option( 'siteurl' ),
+			'gmt_offset'          => get_option( 'gmt_offset' ),
+			'timezone_string'     => get_option( 'timezone_string' ),
+			'stats_version'       => Constants::get_constant( 'STATS_VERSION' ),
+			'stats_api'           => 'jetpack',
+			'page_on_front'       => get_option( 'page_on_front' ),
+			'permalink_structure' => get_option( 'permalink_structure' ),
+			'category_base'       => get_option( 'category_base' ),
+			'tag_base'            => get_option( 'tag_base' ),
+		);
+		$blog = array_merge( Options::get_options(), $blog );
+		unset( $blog['roles'], $blog['blog_id'] );
+
+		add_filter( 'esc_html', array( $this, 'filter_esc_html_check_if_string' ), 10, 2 );
+		$blog = map_deep( $blog, 'esc_html' );
```

Comment:
I'd like to explain this a bit.
Looking at the corresponding [Stats module code](https://github.com/Automattic/jetpack/blob/d38405b569aad92c4164b61acb57b884dc11b2c0/projects/plugins/jetpack/modules/stats.php#L1045) there is the `stats_esc_html_deep` function which I can confirm is not used anywhere outside of that file.
I'd really like to deprecate that function, as most of its functionality (especially the recursive logic) is already covered by `map_deep` core function.
I had to add the filter to make sure boolean values retuned by `get_blog` are not converted to strings, therefore ensure backwards compatibility.

Reply:
Yes, it looks cleaner this way. :)